### DISCUSSION TITLE
Usage of default buffer for tableName

### DISCUSSION THREAD

**Discussion by JayMartin0718** [Riverside-Software/sonar-openedge#961] [General]
This warning is raised when a db buffer is not defined for a table. This rule flags all references including those in the main procedure. While defining a buffer in the main procedure resolves the error, it does not resolve the reason for the error (potential scoping issues). Is it possible to run this check only for internal procedures, functions, and methods?

In the example below, I want customer scoped to the main procedure/program. Defining a buffer does not make sense in this scenario and results in false-positives.

// Main
find first customer no-lock no-error.  <-- Buffer error

run checkCust:
  define buffer custBuf for customer.
  find first custBuf no-lock no-error.
end.

**Comment by gquerret** [comment]
A change has been introduced in 2.15 (released yesterday), that allows you to choose scopes to be skipped.
